Ast endpoints

c-brenn opened the pull request:
* `/api/pml` returns the AST as part of the repsonse - see readme

* `/api/ast` takes an AST and returns the corresponding PML as an attachment

c-brenn committed:
adds ast to `/api/pml` response

c-brenn committed:
adds endpoint for converting an AST to PML

`/api/ast` takes an AST and returns the corresponding PML as an attachment.
The ast must be a valid erlang term, that has been encoded using
`:erlang.term_to_binary` and then base64 encoded for http transmission

c-brenn committed:
adds AST test to Pml ControllerTest

22a committed:
added pml-tx download button

22a commented:
added a pml download button

it looks and works like this:
![2017-03-17 21_13_10](https://cloud.githubusercontent.com/assets/7144173/24063398/5f4385e6-0b57-11e7-91a2-129dbd5f33c7.gif)

22a commented:
The time at which we call [these](https://github.com/tom-and-the-toothfairies/pathways/pull/146/files#diff-af326ae966b05ea64b817eac7636661eR83) is up for debate, the functions were designed in such a way so as to facilitate multiple ast revisions (necessary if PML-TX features are purchased)

houli commented on panacea/web/static/js/app.js:
```diff
@@ -48,7 +48,30 @@ async function handleDrugsResponse(response) {
+  document.body.appendChild(element);
+  element.click();
+  document.body.removeChild(element);
+};
```
Instead of this stuff we can use `Blob` api I believe

houli commented on panacea/web/static/js/app.js:
```diff
@@ -48,7 +48,30 @@ async function handleDrugsResponse(response) {
+
+const prepareDownloadButton = ast => {
+  const pmlDownloadButton = document.getElementById('pml-download-button');
+  pmlDownloadButton.onclick = async function(){
```
addEventListener

houli commented on panacea/web/static/css/app.css:
```diff
@@ -34,3 +34,7 @@
   margin: 40px;
 }
+
+.padded-bottom {
```
Instead of this padded class will using bootstrap `row` bits get us the right padding we want?

houli committed:
Merge branch 'iteration1-2' into ast-endpoints

22a committed:
simplified file download logic

c-brenn committed:
adds support for authorization tokens in the query string.

Auth tokens can be provided in the query string using the `authorization_token`
parameter.

All API endpoints accept both header and url auth. In the case that both methods
are set, then the token from the url is used.

22a committed:
re-added pml-tx download ui

houli committed:
Merge branch 'iteration1-2' into ast-endpoints

22a committed:
refactored download button logic

houli commented on panacea/web/static/js/request.js:
```diff
@@ -1,4 +1,4 @@
-const apiAccessToken = document.getElementById('api-access-token').content;
+export const apiAccessToken = document.getElementById('api-access-token').content;
```
Don't need an export any more if I'm not mistaken

22a committed:
removed unnecessary button tag, tried cucumber tests

houli committed:
Nerf PML download Cucumbers

c-brenn committed:
pml controller sends entire Analysis struct

everything in the Analysis struct needs to be sent to the user, so
there's no need to pull each piece out individually.

Moves the encoding of the AST to the `Analysis` module too.

houli committed:
Away with you!

22a committed:
updated changelog with file download info

houli committed:
Update FEATURES doc
